### PR TITLE
Move the SG and weaver binaries to the Realm package

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Removes the Realm.SourceGenerator package and moves the analyzer inside the regular Realm package. This should have no functional impact, but is a slight improvement in terms of maintainability as we have one fewer package to manage. It also makes it easier for users to understand what our dependencies are.

##  TODO

* [x] Changelog entry

### CODE REVIEW

**Review by papafe — APPROVED** [review]
Looks good to me! If the packaging works fine, it seems a really good idea

File: Realm/Realm/Realm.csproj [hunk 1]
```diff
@@ -33,16 +34,19 @@
...
-    <ProjectReference Include="..\Realm.SourceGenerator\Realm.SourceGenerator.csproj" PrivateAssets="Compile" />
+    <ProjectReference Include="..\Realm.Fody\Realm.Fody.csproj" PrivateAssets="None">
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+    </ProjectReference>
+    <ProjectReference Include="..\Realm.SourceGenerator\Realm.SourceGenerator.csproj" PrivateAssets="Compile">
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+    </ProjectReference>
   </ItemGroup>
-  <ItemGroup>
+  <ItemGroup Label="Package">
```
[the remark this answers is not in the document]
Comment (nirinchev) [reply Realm/Realm/Realm.csproj:44]: Yeah, just for readability - if you collapse all nodes, you get several `ItemGroup` sections with no indication what is for what, so figured I'd add a label to clarify things a little.

File: Tools/SetupUnityPackage/CommandLineOptions/TestOptions.cs [hunk 2]
```diff
@@ -32,7 +32,6 @@ internal class TestOptions : OptionsBase
 
         public override ISet<string> IgnoredDependencies { get; } = new HashSet<string>
         {
-            "Realm.SourceGenerator",
             "Realm.Fody",
```
Comment (fealebenpae) [inline Tools/SetupUnityPackage/CommandLineOptions/TestOptions.cs:35]:
```suggestion
```

**Review by fealebenpae — APPROVED** [review]

File: Tools/SetupUnityPackage/CommandLineOptions/TestOptions.cs (hunk 2 above)
Comment (fealebenpae) [reply Tools/SetupUnityPackage/CommandLineOptions/TestOptions.cs:35]: Understood.

**Pull request merged**